modin/core/dataframe/algebra/binary.py

```diff
@@ -39,7 +48,7 @@ def maybe_compute_dtypes_common_cast(
     ----------
     first : PandasQueryCompiler
         First operand for which the binary operation would be performed later.
-    second : PandasQueryCompiler, list-like or scalar
+    second : PandasQueryCompiler, dict, list, tuple, np.ndarray, str or DtypeObj
```

Comment:
Can `second` be any of the new added dtypes?

Reply:
According to `isinstance` checks in this function - it can.